### PR TITLE
Jocelyn july 18

### PR DESCRIPTION
**Pull request by jociet101**
# July 18 PR

# 👩🏻‍🏫 What does this PR do?

- [x] Implement collection view for upcoming sessions on home tab
- [x] Implement double tap gesture to flip collection view cell content view to view details
- [x] Refactor a bunch of stuff according to feedback

### CODE REVIEW

File: Playmate/Models/PlayerConnection.m [hunk 1]
```diff
@@ -43,7 +44,7 @@ + (void)saveMyConnectionTo:(NSString *)otherObjectId withStatus:(BOOL)areFriends
     
     PFUser *me = [[PFUser currentUser] fetchIfNeeded];
     
-    PlayerConnection *playerConnection = [Constants getPlayerConnectionForUser:me];
+    PlayerConnection *playerConnection = [Helpers getPlayerConnectionForUser:me];
```
Comment (albertomota-codes) [inline Playmate/Models/PlayerConnection.m:47]: 💙

File: Playmate/Utilities/APIManager.m [hunk 2]
```diff
@@ -41,10 +42,10 @@ - (id)init {
 - (void)getSportsListWithCompletion:(void(^)(NSDictionary *list, NSError *error))completion {
     
     AFHTTPSessionManager *manager = [AFHTTPSessionManager manager];
-    NSDictionary *params = @{@"parents_only":@YES};
-    [manager GET:[Constants decathalonSportsListString]  parameters:params progress:nil success:^(NSURLSessionDataTask * task, NSDictionary *list) {
+    [manager GET:[Constants decathalonSportsListString]  parameters:nil progress:nil success:^(NSURLSessionDataTask * task, NSDictionary *list) {
         completion(list, nil);
     } failure:^(NSURLSessionDataTask * task, NSError *error) {
+        [Helpers handleAlert:error withTitle:@"Error" withMessage:nil forViewController:self];
```
Comment (albertomota-codes) [inline Playmate/Utilities/APIManager.m:48]: 💙

File: Playmate/Utilities/Constants.m [hunk 3]
```diff
@@ -365,4 +387,30 @@ + (NSDictionary *)titleAttributes {
...
+                                                    @"playmate_logo_transparent",
+                                                    @"playmate_logo_transparent",
+                                                    @"playmate_logo_transparent",
+                                                    @"playmate_logo_transparent",
+                                                    @"confetti",
+                                                    @"playmate_logo_transparent",
+                                                    @"playmate_logo_transparent",
+                                                    @"playmate_logo_transparent",nil];
+    
+    NSDictionary *sportToImageName = [[NSDictionary alloc] initWithObjects:imageNames forKeys:[Constants sportsList:NO]];
```
Comment (albertomota-codes) [inline Playmate/Utilities/Constants.m:411]: `sportList` could be named `sportListIncludingAll:`

So, when you call the method, you know what is the boolean for.

File: Playmate/View Controllers/Home Tab/HomeViewController.m [hunk 4]
```diff
@@ -46,114 +52,69 @@ - (void)viewDidLoad {
...
-	}
-
 	if ([segue.identifier isEqualToString:@"toCalendar"]) {
 		CalendarViewController *vc = [segue destinationViewController];
 		vc.rawSessionList = self.sessionList;
-	}
-
+    } else if ([segue.identifier isEqualToString:@"homeToUpcomingSessions"]) {
+        UpcomingSessionsViewController *vc = [segue destinationViewController];
+        self.delegate = (id)vc;
```
[the remark this answers is not in the document]
Comment (jociet101) [reply Playmate/View Controllers/Home Tab/HomeViewController.m:116]: It gives warning
`Assigning to 'id<HomeViewControllerDelegate> _Nullable' from incompatible type 'UpcomingSessionsViewController *__strong'`
 if I try to assign without casting

File: Playmate/View Controllers/Home Tab/UpcomingSessionsViewController.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Created by Jocelyn Tseng on 7/18/22.
+//
+
+#import <UIKit/UIKit.h>
+#import <Parse/Parse.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface UpcomingSessionsViewController : UIViewController
```
Comment (albertomota-codes) [inline Playmate/View Controllers/Home Tab/UpcomingSessionsViewController.h:13]: I think, if you add here (remove it from the .m)
you will not have to cast to (id)

**Pull request merged**